File: bench/benchmarks/leibniz/leibniz.c
/* Benchmark: Leibniz series approximation of π. Same algorithm as .monk. */
#include <stdio.h>

int main(void) {
    long N = 50000000;
    double sum = 0.0;
    for (long k = 0; k < N; k++) {
        double term = 1.0 / (double)(2 * k + 1);
        if (k % 2 == 0) {
            sum += term;
        } else {
            sum -= term;
        }
    }
    double pi = 4.0 * sum;
    long scaled = (long)(pi * 10000000.0);
    printf("%ld\n", scaled);
    return 0;
}

File: bench/benchmarks/leibniz/leibniz.go
//go:build ignore

package main

import "fmt"

func main() {
	N := int64(50_000_000)
	sum := 0.0
	for k := int64(0); k < N; k++ {
		term := 1.0 / float64(2*k+1)
		if k%2 == 0 {
			sum += term
		} else {
			sum -= term
		}
	}
	pi := 4.0 * sum
	scaled := int64(pi * 10000000.0)
	fmt.Println(scaled)
}
